dandi/dandiarchive.py

```diff
@@ -48,7 +48,10 @@ def navigate_url(url):
         client = DandiAPIClient(server_url)
         if asset_id["version"] is None:
             r = client.get(f"/dandisets/{asset_id['dandiset_id']}/")
-            asset_id["version"] = r["most_recent_version"]["version"]
+            asset_id["version"] = r["draft_version"]["version"]
+            published_version = r["most_recent_published_version"]
+            if published_version:
+                asset_id["version"] = published_version["version"]
```

Comment:
Let's make it non-breaking so we could even release `dandi-cli` right away:
```suggestion
            if "draft_version" in r:
                asset_id["version"] = r["draft_version"]["version"]
                published_version = r["most_recent_published_version"]
                if published_version:
                    asset_id["version"] = published_version["version"]
            else:
                # TODO: remove `if` after https://github.com/dandi/dandi-api/pull/219 is merged/deployed
                asset_id["version"] = r["most_recent_version"]["version"]            
```